### PR TITLE
Make sequence numbers local to instructions

### PR DESCRIPTION
**Pull request by bjorn3**
This allows renumbering to be localized to a single block where previously it could affect the entire function. Also saves 32bit of overhead per block.

I originally set out to remove sequence numbers entirely to save 32bit of overhead for every instruction, but the dominator tree code uses sequence numbers to find the relative order of instructions within a block.

Builds on top of https://github.com/bytecodealliance/wasmtime/pull/6042. I split both of them to simplify bisection and help with reviewing.

### CODE REVIEW

**Review by cfallin — APPROVED** [review]
This looks good as well -- thanks for finding this simplification opportunity! Happy to merge once #6042 lands and this is rebased.

**Comment by bjorn3** [comment]
Rebased

**Pull request merged**
